src/AgentImplementation.sol

```diff
@@ -309,6 +316,23 @@ contract AgentImplementation is IAgent, ERC721Holder, ERC1155Holder {
         }
     }
 
+    function _charge(DataType.Fee[] memory fees, bytes32 referral, bool payFromAgent) internal {
```

Comment:
That's doable. However it will require an extra action to generate `referral` for feeCollector with 100%. Anther way is that we can define rate 0 as 100%. Then the referral will be simply casting from address to bytes32.

Reply:
How about changing the storage from `address feeCollector` to `bytes32 defaultReferral`, which is set the address with a 100% rate when deployment. It would more closely match the existing `referrals` design and simplify the code in `FeeLibrary.sol`. However, this would mean that any usage of current `feeCollector` would require the `_parse` function to get the address. What do you think?

Reply:
Good idea